syntax = "proto3";

package is;

import "google/protobuf/wrappers.proto";
import "is/msgs/validate.proto";

message SkeletonsDetectorOptions {

  enum Model {
    COCO = 0;
    MPI = 1;
    MPI_4_layers = 2;
  }

  // width and height must be multiples of "16" or "-1" (automatic size)
  message Size {
    int32 width = 1[(is.validate.rules).int32 = {gte: -1}];
    int32 height = 2[(is.validate.rules).int32 = {gte: -1}];
  }

  string broker_uri = 1;
  string zipkin_host = 2;
  uint32 zipkin_port = 3;

  string models_folder = 4;
  Model model = 5;
  // Multiples of 16. If it is increased, the accuracy potentially increases. 
  // If it is decreased, the speed increases. For maximum speed-accuracy balance, 
  // it should keep the closest aspect ratio possible to the images or videos to be
  // processed. Using `-1` in any of the dimensions, OP will choose the optimal 
  // aspect ratio depending on the user's input value. E.g. the default `-1x368`
  // is equivalent to `656x368` in 16:9 resolutions, e.g. full HD (1980x1080) 
  // and HD (1280x720) resolutions.
  Size network_resolution = 6;
  // The image resolution (display and output). Use "-1x-1" to force the
  // program to use the input image resolution.
  Size output_resolution = 7;
  // Number of scales to average.
  google.protobuf.Int32Value scale_number = 8;
  // Scale gap between scales. No effect unless scale_number > 1.
  // Initial scale is always 1. If you want to change the initial scale, 
  // you actually want to multiply the `net_resolution` by your desired initial scale.
  google.protobuf.DoubleValue scale_gap = 9;
  // GPU device start number.
  uint32 num_gpu_start = 10;
}